homeassistant/components/kdeconnect/config_flow.py

```diff
@@ -0,0 +1,153 @@
+"""Config flow for KDEConnect integration."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any, cast
+
+from cryptography.hazmat.primitives import serialization
+from pykdeconnect.client import KdeConnectClient
+from pykdeconnect.const import PairingResult
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_DEVICE_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import (
+    CONF_DEVICE_CERT,
+    CONF_DEVICE_INCOMING_CAPS,
+    CONF_DEVICE_NAME,
+    CONF_DEVICE_OUTGOING_CAPS,
+    CONF_DEVICE_TYPE,
+    CONF_REFRESH,
+    CONNECT_TIMEOUT,
+    DATA_KEY_CLIENT,
+    DATA_KEY_NAMES,
+    DOMAIN,
+)
+from .helpers import ensure_running
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def try_pair(hass: HomeAssistant, device_id: str) -> dict[str, Any]:
```

Comment:
Can you be more specific with the typing rather than `Any`?

Reply:
This returns the data for the ConfigEntry, which home assistant itself types as `Mappinng[str, Any]`. This function is also typed as `dict[str, Any]` in the config_flow template (The function is called validate_input there).

I also checked other integrations. tuya also types this as `dict[str, Any]`, while pihole even uses just `dict` to type it's data.

Should I still use a narrower type here? 
The narrowest type in my case would be `dict[str, str | set[str]]`, but that doesn't feel like useful information, as the dict isn't homogeneous.

A better approach might be using a TypedDict, but I couldn't find precedent for using TypedDict for ConfigEntry data in any other integration.

Reply:
You can cast to the right type at the time of use too.  Typed Dicts are also acceptable.  Dict[str, Any] is hella loose.